File: algorithms-and-data-structures/algorithms-and-data-structures-2/data-structures/linear-data-structures/stack/stack.c
typedef struct{
	int* vet;
	int topo,capacidade;
}stack;
void construtor(stack* pilha, int n){
	(*pilha).topo=-1;
	(*pilha).capacidade=n;
	(*pilha).vet=malloc(n*sizeof(int));
}
void push(*stack pilha,int n){
	if((*pilha).topo>=(*pilha).capacidade){
		return;
	}
	(*pilha).vet[(*pilha).++topo]=n;	
}
int pop(*stack pilha){
	if((*pilha).topo<0){
		return;
	}
	int resp=(*pilha).vet[(*pilha).topo--];
	return resp;
}
int isVazia(*stack pilha){
	if((*pilha).topo==-1){
		return 1;
	}
	else{
		return 0;
	}
}
int size(){
	return (*pilha).topo+1;
}

File: algorithms-and-data-structures/algorithms-and-data-structures-2/data-structures/linear-data-structures/stack/stack.java
class Stack[
	int[] array;
	int topo,capacidade;
	public Stack(int n){
		array=new int[n];
		topo=-1,capacidade=n;
		
	}
	void push(int n){
		if(topo>=capacidade){
			return;
		}
		array[++topo]=n;	
	}
	int pop(){
		if(topo<0){
			return;
		}
		int resp=array[topo--];	
		return resp;
	}
	boolean isVazia(){
		return topo==-1;
	}
	int size(){
		return topo+1;
	}
	
}
